test/topicFilters.spec.ts

```diff
@@ -0,0 +1,296 @@
+      errorWasThrown = true;
+      should.equal(!error.message.search("name is missing"), false);
+      console.log(error.message);
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Adding a rule with no filter", async function(): Promise<void> {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("Priority_1", "");
+    } catch (error) {
+      errorWasThrown = true;
+      should.equal(!error.message.search("Filter is missing"), false);
+      console.log(error.message);
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Adding a rule with a Boolean filter whose input is not Boolean", async function(): Promise<
+    void
+  > {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("Priority_1", "priority = 1");
+      await subscriptionClient.addRule("Priority_2", "1");
+    } catch (error) {
+      errorWasThrown = true;
+    }
+    should.equal(errorWasThrown, true);
+  });
+  /*
+  it("Adding a rule with a SQL action whose input is not a string", async function(): Promise<
+    void
+  > {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule(
+        "Priority_1",
+        "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')",
+        ""
+      );
+    } catch (error) {
+      errorWasThrown = true;
+      // should.equal(!error.message.search("Filter is missing"), false);
+      console.log(error.message);
+    }
+    should.equal(errorWasThrown, true);
+  });*/
+  it("Removing non existing rule on a subscription that doesnt have any rules should throw error", async function(): Promise<
+    void
+  > {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.removeRule("Priority_5");
+    } catch (error) {
+      should.equal(!error.message.search("could not be found"), false);
+      errorWasThrown = true;
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Removing non existing rule on a subscription that has other rules should throw error", async function(): Promise<
+    void
+  > {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("Priority_1", "priority = 1");
+      await subscriptionClient.removeRule("Priority_5");
+    } catch (error) {
+      errorWasThrown = true;
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Subscription with 0/1/multiple rules returns rules as expected", async function(): Promise<
+    void
+  > {
+    let rules = await subscriptionClient.getRules();
+    should.equal(rules.length, 0);
+
+    let expr = "(priority = 1)";
+    await subscriptionClient.addRule("Priority_1", expr);
+    rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+    should.equal(JSON.stringify(rules[0].filter), JSON.stringify({ expression: expr }));
+
+    expr = "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')";
+    await subscriptionClient.addRule(
```

Comment:
Lets make the second rule have an action as well, and then add a check to see if the action actually got added to the rule

Reply:
Action is not getting added eventhough I pass it as an argument.
I checked the filters in the samples, they are working fine.
Tried the action part, action has no effect as is not even getting added.